hal/src/sleeping_delay.rs

```diff
@@ -35,42 +36,46 @@ where
     }
 }
 
-impl<TIM, TYPE> DelayUs<TYPE> for SleepingDelay<TIM>
+impl<TIM, TYPE> ehal_02::blocking::delay::DelayUs<TYPE> for SleepingDelay<TIM>
 where
     TIM: InterruptDrivenTimer,
     TYPE: Into<u32>,
 {
     fn delay_us(&mut self, us: TYPE) {
-        let us: u32 = us.into();
+        <Self as DelayNs>::delay_us(self, us.into());
+    }
+}
+
+impl<TIM, TYPE> ehal_02::blocking::delay::DelayMs<TYPE> for SleepingDelay<TIM>
+where
+    TIM: InterruptDrivenTimer,
+    TYPE: Into<u32>,
+{
+    fn delay_ms(&mut self, ms: TYPE) {
+        <Self as DelayNs>::delay_ms(self, ms.into());
+    }
+}
 
+impl<TIM: InterruptDrivenTimer> DelayNs for SleepingDelay<TIM> {
+    fn delay_ns(&mut self, ns: u32) {
         // Determine how many cycles we need to run for this delay, if any
         // Avoid timers that run longer than a second because for 48 MHz-based timers,
         //   there is no valid divisor + cycle count greater than ~1.3s, so we'd panic.
-        let mut count: u32 = 1 + (us / NUM_US_IN_S);
+        let mut loop_counter: u32 = 1 + (ns / NUM_NS_IN_S);
 
         // Start the timer and sleep!
-        self.timer.start((us / count).nanos());
+        self.timer.start((ns / loop_counter).nanos());
         self.timer.enable_interrupt();
         loop {
             asm::wfi();
             if self.timer.wait().is_ok() || self.interrupt_fired.load(atomic::Ordering::Relaxed) {
                 self.interrupt_fired.store(false, atomic::Ordering::Relaxed);
-                count -= 1;
-                if count == 0 {
+                loop_counter -= 1;
+                if loop_counter == 0 {
                     break;
                 }
             }
         }
         self.timer.disable_interrupt();
     }
 }
```

Comment:
@sajattack, if there's one change I'd to see get tested, it'd be this one. I'm not sure if this implementation is correct.